rust-toolchain.toml

```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "1.79.0"
```

Comment:
Explicitly setting the version might be a bad idea.

Reply:
In what way exactly? IMHO it makes sure all people working on the project use the same version.
Also its read by the setup-rust action.

Reply:
Yes, but there might be an openapi bump in the future, which might need a newer stable.
Just using the latest stable in ci ensures that the ci reflects what the compile result would be, if it was a release.

Reply:
The generator is pinned aswell so I assume that what you describe could not happen. If the decision is made to update, one would have to bump the generator version and this.
Working with a lot of docker images and being burned by the `latest` tag in production, IMHO repoducible builds and consistent versioning outweight the conveniece of just getting the latest version.

Reply:
Actually to be fair, the generate script is not idempotent as it does not pull a specific version of the spec. Which is another issue in of itself as this means one commit which could build at one point time, could fail at some other time.
To elaborate on why I think this is important (arguably contrived as we do not benchmark, work on such a low level that compiler decisions matter a whole lot or distribute compiled binaries, but for the sake of argument lets assume they do): Someone reports an issue with a specific version. Someone then tries to repoduce the issue with a different rust version and cannot do it, cuz the latest version uses a new LLVM version that sidesteps the root cause of the issue cuz it reorders some execution slightlty different. This makes confirming the issue and eventually debugging it becomes harder as you now have to factor in a range of versions the libary could have been compiled against. Assuming everything else is idempotent, when using a version that is tied to a commit allows for a ci build to repeatable and prove that a given commit works against a specific commit.

Reply:
I just don't think that reproducibility is what matters here.
I think that having a ci that just succeeds without any changes is more important.

The codegen being repeatable is also not an issue imo.
If you want the same codegen, you can always grab a specific version of the openapi-spec.
(I did that for #19 for example, to get rid of any extra changes)

Also anyone can compile their binary with any rust version they want. So we are going to have that hypothetical issue with LLVM regardless. As a library, we cannot make 100% repeatable builds. NixOs imo does a pretty good job of doing reproducible builds for rust binaries, but that's the closest you can come to bit-to-bit reproducible builds (and even then builds are sometimes not bit-to-bit reproducible).

Reply:
I did some digging and the toolchain file is mostly used in projects that rely on nightly/unstable features to get some semblance of stability. As none are used, this is not as important. Ill remove it and update the action.